src/actions.rs

```diff
@@ -0,0 +1,641 @@
+                    v2 VARCHAR(128) NOT NULL,
+                    v3 VARCHAR(128) NOT NULL,
+                    v4 VARCHAR(128) NOT NULL,
+                    v5 VARCHAR(128) NOT NULL,
+                    PRIMARY KEY(id),
+                    CONSTRAINT unique_key UNIQUE(ptype, v0, v1, v2, v3, v4, v5)
+                ) ENGINE=InnoDB DEFAULT CHARSET=utf8;",
+    )
+    .execute(&mut conn)
+    .await
+    .map_err(|err| CasbinError::from(AdapterError(Box::new(Error::SqlxError(err)))))
+}
+
+#[cfg(feature = "postgres")]
+pub async fn remove_policy(mut conn: &ConnectionPool, pt: &str, rule: Vec<String>) -> Result<bool> {
+    let rule = normalize_casbin_rule(rule, 0);
+    sqlx::query!(
+        "DELETE FROM casbin_rules WHERE
+                    ptype = $1 AND
+                    v0 = $2 AND
+                    v1 = $3 AND
+                    v2 = $4 AND
+                    v3 = $5 AND
+                    v4 = $6 AND
+                    v5 = $7",
+        pt.to_string(),
+        rule[0],
+        rule[1],
+        rule[2],
+        rule[3],
+        rule[4],
+        rule[5]
+    )
+    .execute(&mut conn)
+    .await
+    .map(|n| n == 1)
+    .map_err(|err| CasbinError::from(AdapterError(Box::new(Error::SqlxError(err)))))
+}
+
+#[cfg(feature = "mysql")]
+pub async fn remove_policy(mut conn: &ConnectionPool, pt: &str, rule: Vec<String>) -> Result<bool> {
+    let rule = normalize_casbin_rule(rule, 0);
+    sqlx::query!(
+        "DELETE FROM casbin_rules WHERE
+                    ptype = ? AND
+                    v0 = ? AND
+                    v1 = ? AND
+                    v2 = ? AND
+                    v3 = ? AND
+                    v4 = ? AND
+                    v5 = ?",
+        pt.to_string(),
+        rule[0],
+        rule[1],
+        rule[2],
+        rule[3],
+        rule[4],
+        rule[5]
+    )
+    .execute(&mut conn)
+    .await
+    .map(|n| n == 1)
+    .map_err(|err| CasbinError::from(AdapterError(Box::new(Error::SqlxError(err)))))
+}
+
+#[cfg(feature = "postgres")]
+pub async fn remove_policies(
+    conn: &ConnectionPool,
+    pt: &str,
+    rules: Vec<Vec<String>>,
+) -> Result<bool> {
+    let mut transaction = conn
+        .begin()
+        .await
+        .map_err(|err| CasbinError::from(AdapterError(Box::new(Error::SqlxError(err)))))?;
+    for rule in rules {
+        let rule = normalize_casbin_rule(rule, 0);
+        sqlx::query!(
+            "DELETE FROM casbin_rules WHERE
+                    ptype = $1 AND
+                    v0 = $2 AND
+                    v1 = $3 AND
+                    v2 = $4 AND
+                    v3 = $5 AND
+                    v4 = $6 AND
+                    v5 = $7",
+            pt.to_string(),
+            rule[0],
+            rule[1],
+            rule[2],
+            rule[3],
+            rule[4],
+            rule[5]
+        )
+        .execute(&mut transaction)
+        .await
+        .and_then(|n| {
+            if n >= 1 {
```

Comment:
here it should be `n==1` it seems, I thought you insert all at one time

Reply:
same as above. fixed.